src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,185 @@
 #include <boost/wave/cpplexer/cpp_lex_token.hpp>
 #include <boost/wave/cpplexer/cpp_lex_iterator.hpp>
 
-#define TCPP_IMPLEMENTATION
-#include <tcpp/source/tcppLibrary.hpp>
-#undef TCPP_IMPLEMENTATION
+//#define TCPP_IMPLEMENTATION
+//#include <tcpp/source/tcppLibrary.hpp>
+//#undef TCPP_IMPLEMENTATION
 
 using namespace nbl;
 using namespace nbl::asset;
 using Microsoft::WRL::ComPtr;
 
 static constexpr const wchar_t* SHADER_MODEL_PROFILE = L"XX_6_7";
 
+
+
 namespace nbl::asset::hlsl::impl
 {
     struct DXC {
         ComPtr<IDxcUtils> m_dxcUtils;
         ComPtr<IDxcCompiler3> m_dxcCompiler;
     };
+
+
+    // for including builtins 
+    struct load_file_or_builtin_to_string
+    {
+        template <typename IterContextT>
+        class inner
+        {
+        public:
```

Comment:
remove the indent for no reason

also keep an indent between `{` and public/protected/private access qualifiers